lib/routes/authorization.js

```diff
@@ -80,17 +81,19 @@ function generateCode(claims, client, scope, req) {
       },
       response: out
     });
+    activityEvent('generateCode', claims.uid, hex(req.payload.client_id), req.headers['user-agent']);
     return out;
   });
 }
 
-function generateGrant(claims, client, scope) {
+function generateGrant(claims, client, scope, req) {
   return db.generateToken({
     clientId: client.id,
     userId: buf(claims.uid),
     email: claims['fxa-verifiedEmail'],
     scope: scope
   }).then(function(token) {
+    activityEvent('generateGrant', claims.uid, hex(client.id), req.headers['user-agent']);
```

Comment:
I'm not sure there's user-facing or metrics-facing value in having separate `generateGrant` and `generateCode` events.  The high-level action that took place here was "authorized a relier", please consider combining these into a single `authorize` event that can be logged further up the chain where you already have the `request` object available.